app/src/main.ts

```diff
@@ -303,85 +282,109 @@ app.on(
 );
 
 async function onReady(): Promise<void> {
-  // Warning: `mainWindow` below is the *global* unique `mainWindow`, created at init time
-  mainWindow = await createMainWindow(appArgs, setDockBadge);
-
-  createTrayIcon(appArgs, mainWindow);
-
-  // Register global shortcuts
-  if (appArgs.globalShortcuts) {
-    appArgs.globalShortcuts.forEach((shortcut) => {
-      globalShortcut.register(shortcut.key, () => {
-        shortcut.inputEvents.forEach((inputEvent) => {
-          // @ts-expect-error without including electron in our models, these will never match
-          mainWindow.webContents.sendInputEvent(inputEvent);
+  //Double check this isn't a second instance when we only want a single instance
+  //Moving the check here (from https://github.com/nativefier/nativefier/blob/8dbe7943d462d1c446a05ebf53d310733661c69b/app/src/main.ts#L255) prevents a second window from briefly showing up before closing
+  if (appArgs.singleInstance && !app.requestSingleInstanceLock()) {
+    app.quit();
+  } else {
+    //This block is run if singleInstance==false or if singleInstance==true and this is the first instance
+    //Since we don't want the primary instance getting focus when a second instance is opened and singleInstance==false we double check singleInstance==true
+    if (appArgs.singleInstance) {
```

Comment:
In retrospect I don't think this check is really needed since 'second-instance' doesn't trigger when not in singleInstance mode, but might be worth keeping it in case that changes in the future?